website/project/decorators.py

```diff
@@ -179,6 +179,14 @@ def check_can_access(node, user, key=None, api_node=None):
     if not node.can_view(Auth(user=user)) and api_node != node:
         if key in node.private_link_keys_deleted:
             status.push_status_message('The view-only links you used are expired.', trust=False)
+        elif node.embargo and node.embargo_end_date is not False:
+            raise HTTPError(
+                http.FORBIDDEN,
+                data={
```

Comment:
@Nesiehr HTTPError is called regardless with the only thing that is changing being the data input into it, just a bit cleaner when done like below:
```
def check_can_access(node, user, key=None, api_node=None):
    """View helper that returns whether a given user can access a node.
    If ``user`` is None, returns False.

    :rtype: boolean
    :raises: HTTPError (403) if user cannot access the node
    """
    if user is None:
        return False
    if not node.can_view(Auth(user=user)) and api_node != node:
        error_data = {
            'message_long': ('User has restricted access to this page. '
            'If this should not have occurred and the issue persists, please report it to '
            '<a href="mailto:support@osf.io">support@osf.io</a>.')
        }
        if key in node.private_link_keys_deleted:
            status.push_status_message('The view-only links you used are expired.', trust=False)
        elif node.embargo and node.embargo_end_date is not False:
            error_data['message_short'] = 'Resource under embargo'
            error_data['message_long'] = 'This resource is currently under embargo, please check back when it opens {}.'.format(node.embargo_end_date.strftime('%A, %b. %d, %Y'))
            }
        raise HTTPError(http.FORBIDDEN, data=error_data)
    return True
```